framework/src/outputs/Checkpoint.C

```diff
@@ -75,60 +77,74 @@ Checkpoint::directory() const
 void
 Checkpoint::output(const ExecFlagType & /*type*/)
 {
-  // Create the output directory
-  std::string cp_dir = directory();
-  Utility::mkdir(cp_dir.c_str());
+  //check periodically if we should be outputting or not
```

Comment:
This comment is a little confusing. I don't think you want to imply how often. If you do want to comment here, you might try to draw attention that you are reading a global variable.